### DISCUSSION TITLE
ADC on NUCLEO-F746ZG with STM32F746

### DISCUSSION THREAD

**Discussion by bevrard59** [micropython/micropython#16377] [STM32 / Pyboard]
Hello, 
if I'm testing this:
```
adc = machine.ADC(machine.Pin('A3'))
val = adc.read_u16()
print(val)
```

No way to receive a value of pin voltage 
Tks for your Help
Benoit

**Comment by robert-hh** [comment]
The voltage depends on the reference value VREF of the ADC. You get it by calculating:

`voltage = VREF * adc.read_u16() / 65536`

VREF is an external signal and provided by the board. If you have a known input voltage, you can conclude back on the value of VREF. Typical values are 3.3V or 1.8V.

**Reply by bevrard59** [reply]
Thanks for your reply but whatever the voltage which is given to the pin the value of adc does not correspond and does not change.I will check if Vref is connected, but for me it is
